framework/scripts/tests/test_wazuh_clusterd.py

```diff
@@ -0,0 +1,362 @@
+            assert task_pool is None
+
+        def start(self):
+            return 'WORKER_START'
+
+    class LocalServerWorkerMock:
+        def __init__(self, performance_test, logger, concurrency_test, node, configuration, enable_ssl, cluster_items):
+            pass
+
+        def start(self):
+            return 'LOCALSERVER_START'
+
+    async def gather(first, second):
+        assert first == 'WORKER_START'
+        assert second == 'LOCALSERVER_START'
+        raise asyncio.CancelledError()
+
+    wazuh_clusterd.cluster_utils = cluster_utils
+    wazuh_clusterd.main_logger = LoggerMock
+    args = Arguments(performance_test='test_performance', concurrency_test='concurrency_test', ssl=True,
+                     send_file=True, send_string=True)
+
+    with patch.object(wazuh_clusterd, 'main_logger') as main_logger_mock:
+        with patch('concurrent.futures.ProcessPoolExecutor', side_effect=FileNotFoundError) as processpoolexecutor_mock:
+            with patch('scripts.wazuh_clusterd.asyncio.gather', gather):
+                with patch('scripts.wazuh_clusterd.logging.info') as logging_info_mock:
+                    with patch('wazuh.core.cluster.worker.Worker', WorkerMock):
+                        with patch('wazuh.core.cluster.local_server.LocalServerWorker', LocalServerWorkerMock):
+                            with pytest.raises(IndexError):
+                                await wazuh_clusterd.worker_main(
+                                    args=args, cluster_config={'test': 'config'},
+                                    cluster_items={'intervals': {'worker': {'connection_retry': 34}}},
+                                    logger='test_logger')
+                            processpoolexecutor_mock.assert_called_once_with(max_workers=1)
+                            main_logger_mock.assert_has_calls([
+                                call.warning(
+                                    "In order to take advantage of Wazuh 4.3.0 cluster improvements, the directory "
+                                    "'/dev/shm' must be accessible by the 'wazuh' user. Check that this file has "
+                                    "permissions to be accessed by all users. Changing the file permissions to 777 "
+                                    "will solve this issue."),
+                                call.warning(
+                                    'The Wazuh cluster will be run without the improvements added in Wazuh 4.3.0 and '
+                                    'higher versions.')
+                            ])
+                            logging_info_mock.assert_called_once_with('Connection with server has been lost. '
+                                                                      'Reconnecting in 10 seconds.')
+                            asyncio_sleep_mock.assert_called_once_with(34)
+
+
+@patch('scripts.wazuh_clusterd.argparse.ArgumentParser')
+def test_get_script_arguments(mock_ArgumentParser):
```

Comment:
```suggestion
def test_get_script_arguments(argument_parser_mock):
```
It would be nice to follow the same structure of naming as in the rest of the test.